inst/include/plant/patch.h

```diff
@@ -105,27 +122,48 @@ class Patch {
 template <typename T, typename E>
 Patch<T,E>::Patch(parameters_type p, environment_type e, Control c)
   : parameters(p),
-    is_resident(p.is_resident),
     environment(e),
-    control(c) {
+    control(c),
+    environment_cache(6) {  // length of ode::Step
   parameters.validate();
 
+  save_RK45_cache = control.save_RK45_cache;
   survival_weighting = p.disturbance;
 
-  // Overwrite all strategy control objects so that they take the
-  // patch control object and also set per species birth rates
-	for (auto i = 0; i < parameters.strategies.size(); ++i) {
-		auto s = parameters.strategies[i];
-    s.control = control;
-    auto spec = Species<T,E>(s);
-    species.push_back(spec);
-  }
+  add_strategies(parameters.strategies);
 
   resource_depletion.reserve(environment.ode_size());
 
   reset();
 }
 
+template <typename T, typename E>
+void Patch<T,E>::overwrite_strategies(std::vector<strategy_type> strategies) {
+  species.clear();
+  add_strategies(strategies);
+}
+
+template <typename T, typename E>
+void Patch<T,E>::add_strategies(std::vector<strategy_type> strategies) {
```

Comment:
I think I know why the many mutants simultaneously is behaving weirdly (only first one returns reasonable values), which is that we never set the birth_rates for the extra mutants! 

Here and in function above `overwrite_strategies` we need to make sure the birth rates are set to 1 for the mutants. Otherwise they're pulling some random number, quite likely 0 but not necessarily.

Reply:
I liked this idea, but I think default `birth_rate` is already set to 1.

```r
# check expanded strategies
map_dbl(lma_strategies_many, ~ .$birth_rate_y)
1 1 1 1 1 1 1 1 1 1

# check species after mutant run
map_dbl(scm$patch$species, ~ .$extrinsic_drivers$evaluate("birth_rate", 1)) # constant through time by default
1 1 1 1 1 1 1 1 1 1
 
signif(scm$net_reproduction_ratios, 1)
3e+02 8e-307 4e-307  0e+00  2e+76 4e-309 1e-310 6e-311 5e-311 6e-311
 ```

Reply:
darn!

Reply:
Possibly still worth adding a print statement to show this rate is being used. 

In any case, print statements are my next suggestion for diagnosing why mutants 2+ are behaving differently to mutant 1. 

Make them all the same strategy and see what's not aligning.